### ISSUE TITLE
Pyright parses `*args` overloads as kwargs?

### ISSUE THREAD

**Issue by Archmonger** [microsoft/pyright#9916]
**Describe the bug**

The following example logically should be valid type hinting syntax, and is supported by other type checkers. However, pyright throws the following errors:
- `Overloaded implementation is not consistent with signature of overload 1`
- `Overload 2 for "example" will never be used because its parameters overlap overload 1`

Here is a code snippet that demonstrates the errors:
```python
@overload
def example(props: dict[str, Any], *children: str) -> None: ...

@overload
def example(*children: str) -> None: ...

def example(*props_and_children: dict[str, Any] | str) -> None: ...
```

But oddly, if the function definition is changed to add kwargs, then pyright believes the first error (`Overloaded implementation is not consistent with signature of overload 1`) resolves itself. However, the second error remains.

For example:
```python
@overload
def example(props: dict[str, Any], *children: str) -> None: ...

@overload
def example(*children: str) -> None: ...

def example(*args: dict[str, Any] | str, **kwargs: Any) -> None: ...
```

**VS Code extension or command-line**
Error occurs on both pyright 1.1.393 and pylance 2025.2.1

**Comment by erictraut** [comment]
I agree that the "overlapping overload" error is incorrect. This is a bug.

The "implementation is not consistent" error is correct and is not a bug. The problem is that overload 1's signature indicates that the function accepts a keyword argument named `props`, but your implementation doesn't define a parameter that accepts this.

```python
# This call is allowed by overload 1 but not by your implementation
example(props=[])
```

You can fix this by specifying that `props` is a positional-only parameter in overload 1.

```python
def example(props: dict[str, Any], /, *children: str) -> None: ...
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.395.
